tests/basics/class_bases.py.exp

```diff
@@ -0,0 +1,17 @@
+(<class 'object'>,)
```

Comment:
Why does the test need this .exp file, does the behaviour not match CPython?  If possible the .exp files should not be added.

Reply:
Unfortunately CPython gives me `<class __main__.A>` with the __main__ module being in the class name. Where as MicroPython does not. So it's actually lines 3 and 4 that are the difference from CPython.

I wasn't sure if the difference from CPython was a lack of my own understanding or an intentional difference from CPython. Either way I decided that it wasn't important to dig deeper in to the cause of the difference since 

1. The test still correctly tests the feature 
2. If MicroPython behaves like CPython in the future this test will break and at that time the .exp file can be removed.

What are your thoughts ?

Reply:
If it's just the printing of a class that differs then we can workaround that in the test, eg by just printing part of the name (see other tests how this is done) or comparing with the class it should be, like `print(result == object)`.

In the end the test should test that the behaviour is the same as CPython (like the returned classes are correct), and not that the printing of classes is the same.

Reply:
> by just printing part of the name

Eg `print(repr(X)[:4])`

Reply:
I just changed those test lines entirely as everything else about the result is checked later. The tests are probably excessive to be honest. But that doesn't hurt, I don't think. 

I had trouble checking at my end as I got errors regarding the MPY version changing. I didn't have the time to debug this so I rebased, pushed, and hope that the CI will sort it.